src/diagnostic/diagnostic_manager.h

```diff
@@ -92,17 +92,17 @@ class DiagnosticsManager : public IDiagnosticsManager
     bool needsWrite(DiagnosticProperties& diag, double timeStamp, double timeStep)
     {
         auto nextWrite = nextWrite_[diag.type + diag.quantity];
-
         return nextWrite < diag.writeTimestamps.size()
-               and std::abs(diag.writeTimestamps[nextWrite] - timeStamp) < timeStep;
+               and static_cast<float>(std::abs(diag.writeTimestamps[nextWrite] - timeStamp))
```

Comment:
can we add a sweet little comment explaining this not so sweet issue we had and pushed us to do this otherwise weird looking static casts?

Reply:
sure thing